cpp/include/raft/matrix/detail/linewise_op.cuh

```diff
@@ -26,6 +27,15 @@ namespace raft {
 namespace matrix {
 namespace detail {
 
+namespace stdex    = std::experimental;
+using extents_type = stdex::extents<stdex::dynamic_extent, stdex::dynamic_extent>;
+template <typename ValueType>
+using padded_layout_row_major =
+  stdex::layout_padded_general<ValueType, stdex::StorageOrderType::row_major_t, 128>;
+template <typename ElementType>
+using padded_matrix =
```

Comment:
The same here (so that the api in `matrix.cuh` doesn't show anything in `detail` namespace).